### DISCUSSION TITLE
Double click to edit

### DISCUSSION THREAD

**Discussion by s2b-git** [YetiForceCompany/YetiForceCRM#15977] [Feature improvement]
How about changing quick edit mode to double click instead of little pencil?
This could apply to list/related list view too. It will be very helpful in improving performance.

**Comment by TonyM15** [comment]
It has been touched few times, but seems like developer dont work on large set of data. In the situation where navigation in detail view has been cut off, and where record preview list works very very slow, this will be absolutely helpfull improvement. 
Maybe at least some comment? We talking about improving UI+UX not about additional functionality

**Comment by ArniSamano** [comment]
I would be excellent to have duble click in list and related list view. For for instance leads generation center like I working in it is essential functionality

**Comment by apcloic** [comment]
Not as straightforward as double click but Occurrences module have a nice feature to edit related Participants status & comment fields from listview using a small modal popup.

![listview_edit_alternative gif](https://user-images.githubusercontent.com/2930560/167657223-b489fd05-79c4-495c-852b-b37f07a13efa.gif)

**Comment by s2b-git** [comment]
This is cool. Perfect for every list items. Is it configurable somewhere? Cannot find here

**Reply by mariuszkrzaczkowski** [reply]
Demo: https://gitdeveloper.yetiforce.com/index.php?module=Occurrences&relatedModule=Contacts&view=Detail&record=1515&mode=showRelatedList&relationId=624&tab_label=LBL_PARTICIPANTS

Doc: https://doc.yetiforce.com/code/classes/Config-Relation.html#property_separateChangeRelationButton

```php
/** Separate change relation button in related module */
public static $separateChangeRelationButton = true;
```
![image](https://user-images.githubusercontent.com/10198654/167780698-cdbba2ae-278e-4177-a413-34090b5f6dee.png)

```php
/** Separate change relation button in related module */
public static $separateChangeRelationButton = false;
```
![image](https://user-images.githubusercontent.com/10198654/167780882-f018e18b-5120-4cd2-a335-3e0e805b36ab.png)

**Reply by wajihjallal** [reply]
Thanks for the reply

**Comment by ArniSamano** [comment]
Having this button in other (including custom) modules will be pretty significant UX improvement. Can we have shortcut to get it in custom way?

Thanks!

**Reply by mariuszkrzaczkowski** [reply]
it won't be that simple, this is a new type of relationship
https://github.com/YetiForceCompany/YetiForceCRM/blob/7a0e0be4ab0479d50535b9ee9c69b4d5bbcc59e1/modules/Occurrences/relations/GetRelatedMembers.php#L1-L131

**Reply by s2b-git** [reply]
> it won't be that simple, this is a new type of relationship
> 
> https://github.com/YetiForceCompany/YetiForceCRM/blob/7a0e0be4ab0479d50535b9ee9c69b4d5bbcc59e1/modules/Occurrences/relations/GetRelatedMembers.php#L1-L131

How about displaying quick edit as a separate button? I am fighting here for efficiency and literally trying to save microseconds, as data which shall be manually edited is rather large.

**Comment by ArniSamano** [comment]
So basically adding record to the table and adusting this file to different module sorting out it... or there are other tricks I didnt noticed?
Thanks for tips
BEst

**Reply by apcloic** [reply]
@ArniSamano 

If you only need the same status / comment fields, you could use the same table (u_#__relations_members_entity) so this is quite easy.

1 - Create relation and adjust in /modules/your_custom_module/relations/GetRelatedMembers.php
2- Rename relation in vtiger_relatedlists table and change from getRelatedList to getRelatedMembers
3- Change $separateChangeRelationButton from 'false' to 'true' in /config/Relation.php

Note : options for 'status_rel' field are defined in vtiger_status_rel table.

**Comment by ArniSamano** [comment]
Clear. Thanks guys

**Comment by ArniSamano** [comment]
The only thing with above mentioned is that it creates additional data. Pain is that I will have to deal with possibly millions of records so changes should stay in the module itself. Still quick edit button as a separate icon would be cool to have

**Comment by s2b-git** [comment]
According to initial thread post,,, is there a plan to upgrade UI in near future to be in line with better UX and actual trends?

Best!

**Reply by mariuszkrzaczkowski** [reply]
Not for the next month, but we're thinking about what to do with it.

**Comment by mariuszkrzaczkowski** [comment]
Added support for double-clicking in the list of records for quick/full editing https://github.com/YetiForceCompany/YetiForceCRM/commit/7da0b553c5dd5f77e851fe37d637fd933ddf0f94  :)

**Reply by s2b-git** [reply]
This is cool. Thanks Marius. Will you merge with gitdev soon?

**Reply by mariuszkrzaczkowski** [reply]
I reloaded gitdev and it's there https://gitdeveloper.yetiforce.com/index.php?module=Leads&view=List&mid=48&parent=47

**Comment by s2b-git** [comment]
Marius, deploying changes directly to last stable does not bring effect. Any additional config is required?

**Reply by mariuszkrzaczkowski** [reply]
hard to say without verification

**Reply by rskrzypczak** [reply]
Disable loading min.js files in config/Developer.php ;)
public static $MINIMIZE_JS = false;

**Reply by s2b-git** [reply]
That was the trick. Respect. This little thing is not that little (for many) 
Thanks

**Reply by garryatbrighton** [reply]
Somehow double-click doesn't work with custom module. What should be done?

Thanks
